deploy/operator/common.sh

```diff
@@ -25,30 +25,17 @@ export ASSISTED_UPGRADE_OPERATOR="${ASSISTED_UPGRADE_OPERATOR:-false}"
 ############
 # Versions #
 ############
-export OPENSHIFT_VERSIONS="${OPENSHIFT_VERSIONS:-$(cat ${__root}/data/default_ocp_versions.json)}"
-
-if [ -z "${ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE:-}" ]; then
-    export ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE=$(echo ${OPENSHIFT_VERSIONS} | jq -rc '[.[].release_image]|max')
-    if [[ "${ASSISTED_UPGRADE_OPERATOR}" == "true" ]]; then
-        # Before the AI operator upgrade, we install the version prior to the most current one of OCP. 
-        # The most current version of OCP we are installing is 4.9, and the version previous to that is 4.8.
-        export ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE=$(echo ${OPENSHIFT_VERSIONS} | jq -rc '[.[].release_image][-2]')
-    fi
-fi
-
-export ASSISTED_OPENSHIFT_VERSION="${ASSISTED_OPENSHIFT_VERSION:-openshift-v4.9.0}"
-
-if [[ "${ASSISTED_UPGRADE_OPERATOR}" == "true" ]]; then
-    VERSION=openshift-v$(echo ${OPENSHIFT_VERSIONS} | jq -rc '[.[].display_name][-2]')
-    export ASSISTED_OPENSHIFT_VERSION="${ASSISTED_OPENSHIFT_VERSION:-${VERSION}}"
+DEFAULT_OS_IMAGES="${DEFAULT_OS_IMAGES:-$(cat ${__root}/data/default_os_images.json)}"
+if [[ "${ASSISTED_UPGRADE_OPERATOR}" == "false" ]]; then
+    RELEASE_IMAGE=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].url]|max')
+    VERSION=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].openshift_version]|max')
+else
+    # Before the AI operator upgrade, we install the version prior to the most current one of OCP. 
+    # The most current version of OCP we are installing is 4.9, and the version previous to that is 4.8.
+    RELEASE_IMAGE=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].url][-2]')
+    VERSION=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].openshift_version][-2]')
 fi
 
-OPENSHIFT_VERSIONS=$(echo ${OPENSHIFT_VERSIONS} |
-    jq -rc 'with_entries(select(.key != "4.6" and .key != "4.7")) | with_entries(
-    {
-        key: .key,
-        value: {rhcos_image:   .value.rhcos_image,
-                rhcos_version: .value.rhcos_version,
-                rhcos_rootfs:  .value.rhcos_rootfs}
-    }
-    )')
\ No newline at end of file
+export OS_IMAGES=$(echo ${DEFAULT_OS_IMAGES} | jq -rc --arg v ${VERSION} 'map(select(.openshift_version==$v))')
+export ASSISTED_OPENSHIFT_VERSION="${ASSISTED_OPENSHIFT_VERSION:-openshift-v${VERSION}}"
+export ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE="${ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE:-${RELEASE_IMAGE}}"
```

Comment:
It will still override versions if user given ``ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE``, right?
Just wanted to make sure

Reply:
Yeah it should still be handled by 'ocp_mirror_release'